### PR TITLE
Refactored search command

### PR DESCRIPTION
**Pull request by guykeller**
## Status
Ready

## Description
1. Improved Splunk's search command readability.
2. Improved search efficiency by an extra 40%.

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

### CODE REVIEW

File: Integrations/SplunkPy/SplunkPy.py [hunk 1]
```diff
@@ -262,82 +262,117 @@ def request(url, message, **kwargs):
...
+    batch_size = int(demisto.args().get("batch_limit", 25000))
+
+    results_offset = 0
+    parsed_search_results = []  # type: List[Dict[str,Any]]
+    dbot_scores = []  # type: List[Dict[str,Any]]
+
+    while len(parsed_search_results) < int(num_of_results) and len(parsed_search_results) < results_limit:
+        current_batch_of_results = get_current_results_batch(search_job, batch_size, results_offset)
+        max_results_to_add = results_limit - len(parsed_search_results)
+        parsed_batch_results, batch_dbot_scores = parse_batch_of_results(current_batch_of_results, max_results_to_add)
```
Comment (ronykoz) [inline Integrations/SplunkPy/SplunkPy.py]: Not sure that this is clear enough

Comment (guykeller) [reply Integrations/SplunkPy/SplunkPy.py]: I agree, I think it's fairly clear but not very clear, 6/10 maybe. Needs improvement.
I can change the naming to raw_results_batch and results_batch, but I could not find any other way to improve it (thought about it a lot - the batching mechanism and the need for efficiency are damaging readability inherently IMHO).
Would love to hear ideas for improvement

Comment (ronykoz) [reply Integrations/SplunkPy/SplunkPy.py]: Lets do this today together, this is not clear enough for me

Comment (guykeller) [reply Integrations/SplunkPy/SplunkPy.py]: Done

**Pull request merged**